app/editor/src/features/content/form/ContentClipForm.tsx

```diff
@@ -0,0 +1,248 @@
+import { FormikForm, FormikText } from 'components/formik';
+import { Modal } from 'components/modal';
+import { IClipModel, IFolderModel, IItemModel } from 'hooks/api-editor';
+import { useModal } from 'hooks/modal';
+import React from 'react';
+import { toast } from 'react-toastify';
+import { Button, ButtonVariant, Col, Row, Text } from 'tno-core';
+
+import { useStorage } from '../../../store/hooks';
+import { defaultFolder } from '../../storage/constants';
+import { ClipDirectoryTable } from './ClipDirectoryTable';
+import { IContentForm } from './interfaces';
+import * as styled from './styled';
+
+export interface IContentClipFormProps {
+  content: IContentForm;
+}
+
+/**
+ * The component to be displayed when the clips tab is selected from the content form.
+ * @returns the ContentClipForm
+ */
+export const ContentClipForm: React.FC<IContentClipFormProps> = ({ content }) => {
+  const videoRef = React.useRef<HTMLVideoElement>(null);
+
+  const [path, setPath] = React.useState('/' + content.source);
+  const [folder, setFolder] = React.useState<IFolderModel>(defaultFolder);
+  const [streamUrl, setStreamUrl] = React.useState<string>();
+  const [item, setItem] = React.useState<IItemModel>();
+  const [start, setStart] = React.useState<string>('');
+  const [end, setEnd] = React.useState<string>('');
+  const [clipNbr, setClipNbr] = React.useState<number>(1);
+  const [currFile, setCurrFile] = React.useState<string>('');
+  const [prefix, setPrefix] = React.useState<string>('');
+  const { toggle, isShowing } = useModal();
+  const storage = useStorage();
+
+  React.useEffect(() => {
+    storage.getFolder(path).then((data) => {
+      setFolder(data);
+    });
+  }, [path, storage]);
+
+  React.useEffect(() => {
+    if (!!streamUrl && !!videoRef.current) {
+      videoRef.current.src = streamUrl;
+    }
+  }, [streamUrl, videoRef]);
+
+  const deleteItem = async (item: IItemModel) => {
+    try {
+      setItem(item);
+      toggle();
+    } catch {
+      // Ignore error a toast would have already been displayed with the error.
+    }
+  };
+
+  const downloadItem = async (item: IItemModel) => {
+    await storage.download(`${folder.path}/${item.name}`);
+  };
+
+  const attachItem = async (item: IItemModel) => {
+    await storage.attach(content.id, `${folder.path}/${item.name}`).then((data) => {
+      toast.success('Attachment added to this snippet.');
+    });
+  };
+
+  const selectItem = (item?: IItemModel) => {
+    setItem(item);
+    setCurrFile(!!item ? item.name : '');
+    setStreamUrl(
+      !!item ? `/api/editor/storage/stream?path=${folder.path}/${item.name}` : undefined,
+    );
+  };
+
+  const createClip = async () => {
+    if (!start || !end) {
```

Comment:
Replace with yup formik validation

Reply:
Worked on this yesterday. Will continue today.

Reply:
We had discussed moving this into a different ticket. I created TNO-418 for this purpose.

Reply:
@jdtoombs